sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1697,11 +1696,17 @@ public PCollection<KafkaRecord<K, V>> expand(PBegin input) {
         }
 
         if (kafkaRead.isRedistributed()) {
-          // fail here instead.
-          checkArgument(
-              kafkaRead.isCommitOffsetsInFinalizeEnabled(),
-              "commitOffsetsInFinalize() can't be enabled with isRedistributed");
+          if (kafkaRead.isCommitOffsetsInFinalizeEnabled()) {
+            LOG.warn(
+                "commitOffsetsInFinalize() will not capture all work processed if set with withRedistribute()");
```

Comment:
How about a little more detailed

Offsets committed due to usage to commitOffsetsInFinalize() may not capture all work processed due to use of withRedistribute()

Reply:
Done.